avidemux_plugins/ADM_videoFilters6/zoom/qt4/Q_zoom.cpp

```diff
@@ -0,0 +1,200 @@
+/***************************************************************************
+                          DIA_zoom.cpp  -  description
+                             -------------------
+
+			    GUI for zooming
+
+    begin                : Fri May 3 2002
+    copyright            : (C) 2002/2007 by mean
+    email                : fixounet@free.fr
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "Q_zoom.h"
+#include "ADM_toolkitQt.h"
+
+//
+//	Video is in YV12 Colorspace
+//
+//
+Ui_zoomWindow::Ui_zoomWindow(QWidget* parent, zoom *param,ADM_coreVideoFilter *in) : QDialog(parent)
+  {
+    uint32_t width,height;
+        ui.setupUi(this);
+        lock=0;
+        // Allocate space for green-ised video
+        width=in->getInfo()->width;
+        height=in->getInfo()->height;
+
+        canvas=new ADM_QCanvas(ui.graphicsView,width,height);
+        
+        myFly=new flyZoom( this,width, height,in,canvas,ui.horizontalSlider);
+        myFly->left=param->left;
+        myFly->right=param->right;
+        myFly->top=param->top;
+        myFly->bottom=param->bottom;
+        myFly->_cookie=&ui;
+        myFly->addControl(ui.toolboxLayout);
+        myFly->upload();
+        myFly->sliderChanged();
+
+
+        connect( ui.horizontalSlider,SIGNAL(valueChanged(int)),this,SLOT(sliderUpdate(int)));
+        connect( ui.pushButtonReset,SIGNAL(clicked(bool)),this,SLOT(reset(bool)));
+#define SPINNER(x) connect( ui.spinBox##x,SIGNAL(valueChanged(int)),this,SLOT(valueChanged(int))); 
+          SPINNER(Left);
+          SPINNER(Right);
+          SPINNER(Top);
+          SPINNER(Bottom);
+
+        show();
```

Comment:
I would prefer not to show() a dialog in ctor but handle all necessary proceedings in showEvent() which is then invoked by exec(). At least I would want to understand the reasons why it was necessary to do it differently here.

Reply:
This filter was made with the help of copy-paste-merge technology© :)
This is especially true for the qt4 part, if i remember correctly, i did not change a thing (its from crop filter)

Reply:
Indeed, I noticed that you kept the qt4 version just the moment I hit "send". The crop filter needs a fix then.

I would not object leaving qt4 out. I didn't try to build Avidemux with Qt4 even once since many months. It is much more important to verify that everything works with qt6.